src/main/java/rx/internal/operators/OnSubscribeSwitchIfEmpty.java

```diff
@@ -69,14 +82,33 @@ public void onCompleted() {
             if (!empty) {
                 child.onCompleted();
             } else if (!child.isUnsubscribed()) {
-                subscribeToAlternate();
+                active = false;
+                subscribe(null);
             }
         }
 
-        private void subscribeToAlternate() {
-            AlternateSubscriber<T> as = new AlternateSubscriber<T>(child, arbiter);
-            serial.set(as);
-            alternate.unsafeSubscribe(as);
+        void subscribe(Observable<? extends T> source) {
+            if (wip.getAndIncrement() == 0) {
+                do {
+                    if (child.isUnsubscribed()) {
+                        break;
+                    }
+
+                    if (!active) {
+                        if (secondary) {
```

Comment:
On a second thought, you are right. If I set `source = null` after the `source.unsafeSubscribe()` the next round will properly go into the alternative. Updated the PR.